pkg/cmd/trait.go

```diff
@@ -206,48 +168,36 @@ func AddTraitDetachCommands(parentCmd *cobra.Command, c types.Args, ioStreams cm
 	return nil
 }
 
-func (o *commandOptions) DetachTrait(cmd *cobra.Command, args []string, ctx context.Context) error {
-	argsLength := len(args)
-	if argsLength < 1 {
-		return errors.New("please specify the name of the app")
-	}
-	c := o.Client
-	namespace := o.Env.Namespace
-
-	var appName = args[0]
-	if err := c.Get(ctx, client.ObjectKey{Namespace: o.Env.Namespace, Name: appName}, &o.AppConfig); err != nil {
+func (o *commandOptions) DetachTrait(cmd *cobra.Command, args []string) error {
+	if err := o.Prepare(cmd, args); err != nil {
 		return err
 	}
-
-	apiVersion, kind, err := cmdutil.GetTraitApiVersionKind(ctx, c, namespace, o.TraitAlias)
+	app, err := application.Load(o.Env.Name, o.appName)
 	if err != nil {
 		return err
 	}
-	for i, com := range o.AppConfig.Spec.Components {
-		if com.ComponentName != appName {
-			continue
-		}
-		var traits []corev1alpha2.ComponentTrait
-		for _, tr := range com.Traits {
-			a, k := tr.Trait.Object.GetObjectKind().GroupVersionKind().ToAPIVersionAndKind()
-			if a == apiVersion && k == kind {
-				continue
-			}
-			traits = append(traits, tr)
-		}
-		o.AppConfig.Spec.Components[i].Traits = traits
+	var traitType = o.Template.Name
+	if err = app.RemoveTrait(o.workloadName, traitType); err != nil {
+		return err
 	}
-	return nil
+	return app.Save(o.Env.Name, o.appName)
 }
 
 func (o *commandOptions) Run(cmd *cobra.Command, ctx context.Context) error {
 	if o.Detach {
-		o.Info("Detaching trait from app", o.Component.Name)
+		o.Infof("Detaching %s from app %s\n", o.Template.Name, o.workloadName)
 	} else {
-		o.Info("Applying trait for app", o.Component.Name)
+		o.Infof("Adding %s for app %s \n", o.Template.Name, o.workloadName)
+	}
+	staging, err := strconv.ParseBool(cmd.Flag(Staging).Value.String())
```

Comment:
Bool flag can be directly retrieved in this way `staging, err := cmd.Flags().GetBool(Staging)`.

Reply:
cool